Error in TFBertForSequenceClassification

## 🐛 Bug

<!-- Important information -->

Model I am using (Bert, XLNet....): Bert

Language I am using the model on (English, Chinese....): Multi-lingual

The problem arise when using:
* [x] the official example scripts: (give details)
* [ ] my own modified scripts: (give details)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details)

<!-- If you have a code sample, error messages, stack traces, please provide it here as well. -->

## Expected behavior

I have fine-tuned a language model using `run_lm_finetuning.py`.

When trying to load it with TFBertForSequenceClassification however, it fails.

```
config = transformers.BertConfig.from_json_file('./bertlm_model/config.json')
model = transformers.TFBertForSequenceClassification.from_pretrained('./bertlm_model/', from_pt = True)
```

Showing the following error:

```
>>> model = transformers.TFBertForSequenceClassification.from_pretrained('./bertlm_model/', from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: classifier.weight not found in PyTorch model
```


If I try to run either `transformers.BertForSequenceClassification.from_pretrained('bertlm_model')` or `transformers.TFBertModel.from_pretrained('bertlm_model', from_pt = True)` all is fine!



<!-- A clear and concise description of what you expected to happen. -->

## Environment

* OS: Ubuntu 18.04
* Python version: 3.7.5
* PyTorch version: 1.3.1
* Transformers version (or branch): Git repo master comit 0cb163865a4c761c226b151283309eedb2b1ca4d
* Using GPU: Yes
* Distributed of parallel setup ?
* Any other relevant information:

The code line that loads the BERT configuration is surely correct:
```
> config = transformers.BertConfig.from_json_file('./bertlm_model/config.json')
```
But, for what concern the loading of a fine-tuned BERT model on a custom dataset, I think it's not correct the line you've used. Can you try with the following line suggested by me?
```
> from transformers import TFBertForSequenceClassification
> model = TFBertForSequenceClassification.from_pretrained('bertlm_model', from_pt = True)
```

I suspect that it doesn't work however. It's a PyTorch->TF 2.0 conversion problem. It would be useful to understand that this bug occurs with only BERT model or with other models.

> ##  Bug
> Model I am using (Bert, XLNet....): Bert
> 
> Language I am using the model on (English, Chinese....): Multi-lingual
> 
> The problem arise when using:
> 
> * [x]  the official example scripts: (give details)
> * [ ]  my own modified scripts: (give details)
> 
> The tasks I am working on is:
> 
> * [ ]  an official GLUE/SQUaD task: (give the name)
> * [x]  my own task or dataset: (give details)
> 
> ## Expected behavior
> I have fine-tuned a language model using `run_lm_finetuning.py`.
> 
> When trying to load it with TFBertForSequenceClassification however, it fails.
> 
> ```
> config = transformers.BertConfig.from_json_file('./bertlm_model/config.json')
> model = transformers.TFBertForSequenceClassification.from_pretrained('./bertlm_model/', from_pt = True)
> ```
> 
> Showing the following error:
> 
> ```
> >>> model = transformers.TFBertForSequenceClassification.from_pretrained('./bertlm_model/', from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: classifier.weight not found in PyTorch model
> ```
> 
> If I try to run either `transformers.BertForSequenceClassification.from_pretrained('bertlm_model')` or `transformers.TFBertModel.from_pretrained('bertlm_model', from_pt = True)` all is fine!
> 
> ## Environment
> * OS: Ubuntu 18.04
> * Python version: 3.7.5
> * PyTorch version: 1.3.1
> * Transformers version (or branch): Git repo master comit [0cb1638](https://github.com/huggingface/transformers/commit/0cb163865a4c761c226b151283309eedb2b1ca4d)
> * Using GPU: Yes
> * Distributed of parallel setup ?
> * Any other relevant information:

Thanks for your answer - unfortunately it didn't work..

As I'm fine-tuning the LM on bert-multilingual, I can't try it out with other models. However I have tried to load all the different BERT huggingface-sub-models using my fine-tuned language model and it seems it is only TFBertModel and TFBertForMaskedLM it will load?

Hope that can lead you in a direction?


```
import transformers
model_dir = 'bertlm_model/'
config = transformers.BertConfig.from_json_file(model_dir + 'config.json')
```

### TFBertModel (works fine)
```
>>> model = transformers.TFBertModel.from_pretrained(model_dir, from_pt = True, config = config)
>>> 
```

### TFBertForPreTraining (won't load)
```
>>> model = transformers.TFBertForPreTraining.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: cls.seq_relationship.weight not found in PyTorch model
>>> 
```

### TFBertForMaskedLM (works fine)
```
>>> model = transformers.TFBertForMaskedLM.from_pretrained(model_dir, from_pt = True, config = config)
>>> 
```

### TFBertForNextSentencePrediction (won't load)
```
>>> model = transformers.TFBertForNextSentencePrediction.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: cls.seq_relationship.weight not found in PyTorch model
>>> 
```

### TFBertForSequenceClassification (won't load)
```
>>> model = transformers.TFBertForSequenceClassification.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: classifier.weight not found in PyTorch model
>>> 
```

### TFBertForMultipleChoice (won't load)
```
>>> model = transformers.TFBertForMultipleChoice.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 109, in load_pytorch_weights_in_tf2_model
    tfo = tf_model(tf_inputs, training=False)  # Make sure model is built
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 822, in __call__
    outputs = self.call(cast_inputs, *args, **kwargs)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_bert.py", line 943, in call
    seq_length = shape_list(input_ids)[2]
IndexError: list index out of range
>>> 
```

### TFBertForTokenClassification (won't load)
```
>>> model = transformers.TFBertForTokenClassification.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: classifier.weight not found in PyTorch model
>>> 
```

### TFBertForQuestionAnswering (won't load)
```
>>> model = transformers.TFBertForQuestionAnswering.from_pretrained(model_dir, from_pt = True, config = config)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
    return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
    return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
  File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
    assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
AssertionError: qa_outputs.weight not found in PyTorch model
>>> 
```

The same pattern of working (e.g. _TFBertForMaskedLM_) vs not working (e.g. _TFBertForQuestionAnswering_) appears also with the PyTorch version of these models? e.g. _BertForMaskedLM_

> Thanks for your answer - unfortunately it didn't work..
> 
> As I'm fine-tuning the LM on bert-multilingual, I can't try it out with other models. However I have tried to load all the different BERT huggingface-sub-models using my fine-tuned language model and it seems it is only TFBertModel and TFBertForMaskedLM it will load?
> 
> Hope that can lead you in a direction?
> 
> ```
> import transformers
> model_dir = 'bertlm_model/'
> config = transformers.BertConfig.from_json_file(model_dir + 'config.json')
> ```
> 
> ### TFBertModel (works fine)
> ```
> >>> model = transformers.TFBertModel.from_pretrained(model_dir, from_pt = True, config = config)
> >>> 
> ```
> 
> ### TFBertForPreTraining (won't load)
> ```
> >>> model = transformers.TFBertForPreTraining.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: cls.seq_relationship.weight not found in PyTorch model
> >>> 
> ```
> 
> ### TFBertForMaskedLM (works fine)
> ```
> >>> model = transformers.TFBertForMaskedLM.from_pretrained(model_dir, from_pt = True, config = config)
> >>> 
> ```
> 
> ### TFBertForNextSentencePrediction (won't load)
> ```
> >>> model = transformers.TFBertForNextSentencePrediction.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: cls.seq_relationship.weight not found in PyTorch model
> >>> 
> ```
> 
> ### TFBertForSequenceClassification (won't load)
> ```
> >>> model = transformers.TFBertForSequenceClassification.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: classifier.weight not found in PyTorch model
> >>> 
> ```
> 
> ### TFBertForMultipleChoice (won't load)
> ```
> >>> model = transformers.TFBertForMultipleChoice.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 109, in load_pytorch_weights_in_tf2_model
>     tfo = tf_model(tf_inputs, training=False)  # Make sure model is built
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/tensorflow_core/python/keras/engine/base_layer.py", line 822, in __call__
>     outputs = self.call(cast_inputs, *args, **kwargs)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_bert.py", line 943, in call
>     seq_length = shape_list(input_ids)[2]
> IndexError: list index out of range
> >>> 
> ```
> 
> ### TFBertForTokenClassification (won't load)
> ```
> >>> model = transformers.TFBertForTokenClassification.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: classifier.weight not found in PyTorch model
> >>> 
> ```
> 
> ### TFBertForQuestionAnswering (won't load)
> ```
> >>> model = transformers.TFBertForQuestionAnswering.from_pretrained(model_dir, from_pt = True, config = config)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_utils.py", line 288, in from_pretrained
>     return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 82, in load_pytorch_checkpoint_in_tf2_model
>     return load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=tf_inputs, allow_missing_keys=allow_missing_keys)
>   File "/home/myuser/.local/share/virtualenvs/BERTlm-giwpujkO/lib/python3.7/site-packages/transformers/modeling_tf_pytorch_utils.py", line 145, in load_pytorch_weights_in_tf2_model
>     assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
> AssertionError: qa_outputs.weight not found in PyTorch model
> >>> 
> ```

--- a/transformers/modeling_tf_pytorch_utils.py
+++ b/transformers/modeling_tf_pytorch_utils.py
@@ -143,7 +143,11 @@ def load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=None, a
         name, transpose = convert_tf_weight_name_to_pt_weight_name(sw_name, start_prefix_to_remove=start_prefix_to_remove)
 
         # Find associated numpy array in pytorch model state dict
-        assert name in pt_state_dict, "{} not found in PyTorch model".format(name)
+        if name not in pt_state_dict:
+            if allow_missing_keys:
+                continue
+            raise AttributeError("{} not found in PyTorch model".format(name))
+
         array = pt_state_dict[name].numpy()
 
         if transpose:
@@ -250,6 +254,7 @@ def load_tf2_weights_in_pytorch_model(pt_model, tf_weights, allow_missing_keys=F
 
     all_tf_weights = set(list(tf_weights_map.keys()))
     loaded_pt_weights_data_ptr = {}
+    missing_keys_pt = []
     for pt_weight_name, pt_weight in current_pt_params_dict.items():
         # Handle PyTorch shared weight ()not duplicated in TF 2.0
         if pt_weight.data_ptr() in loaded_pt_weights_data_ptr:
@@ -258,7 +263,10 @@ def load_tf2_weights_in_pytorch_model(pt_model, tf_weights, allow_missing_keys=F
 
         # Find associated numpy array in pytorch model state dict
         if pt_weight_name not in tf_weights_map:
-            raise ValueError("{} not found in TF 2.0 model".format(pt_weight_name))
+            if allow_missing_keys:
+                missing_keys_pt.append(pt_weight_name)
+                continue
+            raise AttributeError("{} not found in TF 2.0 model".format(pt_weight_name))
 
         array, transpose = tf_weights_map[pt_weight_name]
 
@@ -283,6 +291,7 @@ def load_tf2_weights_in_pytorch_model(pt_model, tf_weights, allow_missing_keys=F
         all_tf_weights.discard(pt_weight_name)
 
     missing_keys, unexpected_keys = pt_model.load_state_dict(new_pt_params_dict, strict=False)
+    missing_keys += missing_keys_pt
 
     if len(missing_keys) > 0:
         logger.info("Weights of {} not initialized from TF 2.0 model: {}".format(

--- a/transformers/modeling_tf_utils.py
+++ b/transformers/modeling_tf_utils.py
@@ -297,7 +297,7 @@ class TFPreTrainedModel(tf.keras.Model):
 
         if from_pt:
             # Load from a PyTorch checkpoint
-            return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file)
+            return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file, allow_missing_keys=True)
 
         ret = model(model.dummy_inputs, training=False)  # build the network with dummy inputs
 

